packages/backend-plugin-api/src/services/system/types.ts

```diff
@@ -263,13 +295,13 @@ export function createServiceFactory<
       ) => PluginServiceFactoryOptions<TService, TContext, TImpl, TDeps>)
     | (() => RootServiceFactoryOptions<TService, TImpl, TDeps>)
     | (() => PluginServiceFactoryOptions<TService, TContext, TImpl, TDeps>),
-): (options: TOpts) => ServiceFactory {
+): ServiceFactoryCompat<TService, 'root' | 'plugin', TOpts> {
   const configCallback =
     typeof options === 'function' ? options : () => options;
   const factory = (
-    o: TOpts,
+    o?: TOpts,
   ): InternalServiceFactory<TService, 'plugin' | 'root'> => {
-    const anyConf = configCallback(o);
+    const anyConf = configCallback(o!);
```

Comment:
👀 , just to confirm, is it safe?

Reply:
Yep, just didn't feel like fully figuring our the types here since it's all going away anyway 😅